apps/onboarding-functions/src/main/java/it/pagopa/selfcare/onboarding/workflow/WorkflowExecutor.java

```diff
@@ -64,6 +64,38 @@ default Optional<OnboardingStatus> onboardingCompletionActivity(TaskOrchestratio
         return Optional.of(OnboardingStatus.COMPLETED);
     }
 
+    default Optional<OnboardingStatus> onboardingCompletionActivity(TaskOrchestrationContext ctx, Onboarding onboarding, boolean sendMail) {
+        final String onboardingString = getOnboardingString(objectMapper(), onboarding);
+
+        //CreateInstitution activity return an institutionId that is used by CreateOnboarding activity
+        String institutionId = ctx.callActivity(CREATE_INSTITUTION_ACTIVITY, onboardingString, optionsRetry(), String.class).await();
+        onboarding.getInstitution().setId(institutionId);
+        final String onboardingWithInstitutionIdString = getOnboardingString(objectMapper(), onboarding);
+
+        ctx.callActivity(CREATE_ONBOARDING_ACTIVITY, onboardingWithInstitutionIdString, optionsRetry(), String.class).await();
+
+        // Create onboarding for test environments if exists (ex. prod-interop-coll)
+        if(Objects.nonNull(onboarding.getTestEnvProductIds()) && !onboarding.getTestEnvProductIds().isEmpty()) {
+            // Schedule each task to run in parallel
+            List<Task<String>> parallelTasks = onboarding.getTestEnvProductIds().stream()
+                    .map(testEnvProductId -> onboardingStringWithTestEnvProductId(testEnvProductId, onboardingWithInstitutionIdString))
+                    .map(onboardingStringWithTestEnvProductId -> ctx.callActivity(CREATE_ONBOARDING_ACTIVITY, onboardingStringWithTestEnvProductId, optionsRetry(), String.class))
+                    .collect(Collectors.toList());
+
+            // Wait for all tasks to complete
+            ctx.allOf(parallelTasks).await();
+        }
+
+        if(sendMail)
+            ctx.callActivity(SEND_MAIL_COMPLETION_ACTIVITY, onboardingWithInstitutionIdString, optionsRetry(), String.class).await();
+
+        return Optional.of(OnboardingStatus.COMPLETED);
+    }
+
+    default Optional<OnboardingStatus> onboardingCompletionActivity(TaskOrchestrationContext ctx, Onboarding onboarding) {
+        return this.onboardingCompletionActivity(ctx, onboarding, true);
+    }
+
```

Comment:
you could avoid to duplicate the method above....you could 
* create an unique method with common logic such as create institution and onboarding
* two method "onboardingCompletionActivity" and "onboardingCompletionActivityWithoutMail" that use common method and in the last calling SEND_MAIL_COMPLETION_ACTIVITY